xiplot/utils/auxiliary.py

```diff
@@ -0,0 +1,63 @@
+from typing import Union
+
+import pandas as pd
+
+CLUSTER_COLUMN_NAME = "Xiplot_cluster"
+SELECTED_COLUMN_NAME = "Xiplot_selected"
+
+
+def get_clusters(aux: pd.DataFrame, n: int = -1) -> pd.Categorical:
+    """Get the cluster column from the auxiliary data.
+
+    Args:
+        aux: Auxiliary data frame.
+        n: Column size if missing. Defaults to `aux.shape[0]`.
+
+    Returns:
+        Categorical column with clusters (creates a column with `n` "all" if missing)
+    """
+    if not isinstance(aux, pd.DataFrame):
+        aux = decode_aux(aux)
+    if CLUSTER_COLUMN_NAME in aux:
+        return pd.Categorical(aux[CLUSTER_COLUMN_NAME].copy())
+    if n == -1:
```

Comment:
Could this be made into an `Optional[int]` instead?

Reply:
You're right, fixed